### ISSUE TITLE
Kink in same coloured branch lines on same line makes them impossible to follow

### ISSUE THREAD

**Issue by mikebeaton** [ForkIssues/TrackerWin#1676]
Fork is excellent software and I use it all the time - thank you very much.

So, this may be a duplicate of my own issue #183, and it's related to #308, but I am still occasionally getting branch topology and colouring which is basically impossible to follow by eye.

Here is an example:

![image](https://user-images.githubusercontent.com/11946605/196958359-2f330d0a-9b46-4daa-ac5d-e0f8ff67b81b.png)

`feature/TAU-118` joins `develop` (via a more-or-less horizontal green line at the bottom if its branch line), and the other green branch of the identical colour (they are exactly the same colour, 0x64da38, I checked) that comes down the middle of the tree crosses over. But I would say it is basically impossible to tell which of these two is which, and I spent ages trying to track down what was going on in my code on the assumption that the green lines were not crossed, i.e. that TAU-118 carries on down after passing very close to the green and brown lines.

Obviously the specific problem here is that the kink in both branches occurs on the same line, and they are of the exact same colour (but it would also be a problem if they were just hard-to-distinguish colours, ofc).

Thank you.

**Comment by DanPristupov** [comment]
How can we improve that?
Adding more colors will not solve the problem because collisions will still happen.

**Comment by mikebeaton** [comment]
I guess I was assuming that maybe you could spot when two uses of the same colour have a kink on the same line and avoid it. I think you're saying that you can't in principle avoid this possibly happening?

If not, perhaps you can make lines which are only changing column go as steeply as possible, whereas lines which are going across to join another branch go as shallowly as possible, so that it is more obvious there there is a crossing; i.e. make the middle green line in the image follow a path very (very!) roughly more like the added line in this picture?

![196958359-2f330d0a-9b46-4daa-ac5d-e0f8ff67b81b](https://user-images.githubusercontent.com/11946605/197044115-7623db75-8201-4da8-91d3-3bfd42b0d607.png)

**Comment by Vankog** [comment]
I agree with the issuer. I had many occasions where it was literally impossible for me to follow the paths, because I could not see which line to follow or where it originates.

Some kind of spacing, bridging or intersection handling would greatly improve this. (e.g. see various diagram tools out there)

Current example I just had today:
![indistinguishable paths](https://user-images.githubusercontent.com/3804909/197739438-0297a135-af73-4818-aaca-d55b409097ae.png)
Sure, if you look REALLY hard you can fiddle and find it out. But it is quite hard.

For comparison, here's the same graph from Tortoise Git:
![distinguishable paths - tortoise](https://user-images.githubusercontent.com/3804909/197740902-d95a9b22-9e57-4b7f-8e58-550059c76b80.png)
and Gitlab:
![distinguishable paths - gitlab](https://user-images.githubusercontent.com/3804909/197740938-2a847cd9-cbf7-4728-ab17-dad85b772192.png)

**Comment by conanthedev** [comment]
Maybe some sort of highlighting/thickening on mouse hover?

**Comment by Pea-ACS** [comment]
I like the highlighting/thickening idea

**Comment by Pea-ACS** [comment]
> How can we improve that? Adding more colors will not solve the problem because collisions will still happen.

Maybe but at least allow us to have more colours and also be able to choose which colours to use in which order. If not through the GUI then at least through a config file

Perhaps allow us to also set line style like "dotted" and "dash", etc

**Comment by KazzmanK** [comment]
May be something like that ?

![image](https://user-images.githubusercontent.com/13328513/200898905-81c9473a-0f4d-435b-a819-094314907c55.png)

**Comment by KazzmanK** [comment]
Also, it can be made clickable, and on click can become bold or default black

**Comment by mikebeaton** [comment]
I have another screenshot. I use Fork multiple times daily, but it is _tricky_ - well, basically, impossible - to follow the branch lines, when there are many overlapping.

In this example, the parent of `previous-day-js` (red arrow) is `Fix the issues related to assessment score section` (bold). I am not sure that it is possible to know this, from the current layout.

![image](https://github.com/fork-dev/TrackerWin/assets/11946605/e5eae2a9-a0ba-4ff8-a187-72047bade7c7)
